### PR TITLE
[mono][interp] Allow passing vtypes with a single scalar field to nat…

### PR DESCRIPTION
**Pull request by vargaz**
…ive code using the faster code path.

This affects types like ObjectHandleOnStack which are passed to icalls.

### CODE REVIEW

**Comment by vargaz** [comment]
This will hopefully fix some of the perf regressions in
https://github.com/dotnet/runtime/issues/79607

File: src/mono/mono/mini/interp/transform.c [hunk 1]
```diff
@@ -2593,6 +2593,40 @@ interp_transform_internal_calls (MonoMethod *method, MonoMethod *target_method,
...
+	int nfields = 0;
+	field = NULL;
+	while ((field = mono_class_get_fields_internal (klass, &iter))) {
+		if (field->type->attrs & FIELD_ATTRIBUTE_STATIC)
+			continue;
+		nfields ++;
+		if (nfields > 1)
+			return FALSE;
+		MonoType *t = mini_get_underlying_type (field->type);
+		if (!((MONO_TYPE_IS_PRIMITIVE (t) || MONO_TYPE_IS_REFERENCE (t) || MONO_TYPE_IS_POINTER (t))))
```
Comment (BrzVlad) [inline src/mono/mono/mini/interp/transform.c:2623]: This seems to incorrectly allow some 8 byte primitives on 32bit targets

**Comment by vargaz** [comment]
On most platform, passing scalar vtypes is done by value, so this should work. If it doesn't, we can ifdef it for the platforms where it works.

**Comment by vargaz** [comment]
The interpreter test failures are relevant.

File: src/mono/mono/mini/interp/transform.c (hunk 1 above)
Comment (BrzVlad) [reply src/mono/mono/mini/interp/transform.c:2623]: I'm not sure I understand how my observation is addressed here. Let's say on 32 bit system, you have to pass a struct that contains just a `long` field. This check here will consider it as valid to pass it as a PTR. But in `do_icall` in `interp.c`, we can only pass a 32bit native int by value.

**Comment by BrzVlad** [comment]
we should probably have consistency between the primitive types that are considered to be passable as a native int and the allowed types of the struct field.

File: src/mono/mono/mini/interp/transform.c (hunk 1 above)
Comment (vargaz) [reply src/mono/mono/mini/interp/transform.c:2623]: There is a size > SIZEOF_VOID_P check at the beginning.

**Review by BrzVlad — APPROVED** [review]

**Comment by vargaz** [comment]
Failures are unrelated

**Pull request merged**